src/main/java/core/basesyntax/model/impl/ReturnOperation.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.model.impl;
+
+import core.basesyntax.exception.InvalidOperationException;
+import core.basesyntax.model.AbstractOperation;
+import core.basesyntax.model.entities.Product;
+import java.util.Map;
+
+public class ReturnOperation<T extends Product> extends AbstractOperation<T> {
+    public ReturnOperation(Map<T, Integer> storage) {
+        super(storage);
+    }
+
+    @Override
+    public void execute(T product, Integer amount) {
+        validateExecution(product, amount);
+        storage.merge(product, amount, Integer::sum);
+    }
+
+    private void validateExecution(T product, Integer amount) {
+        if (!storage.containsKey(product) || storage.get(product) - amount < 0 || amount <= 0) {
```

Comment:
```suggestion
        if (!storage.containsKey(product) || amount <= 0) {
```

Reply:
We cannot return more than we have bought

Reply:
storage.get(product) shows the actual quantity - we don't know how many was bought